### PR TITLE
feat(styles): add the radio/checkbox card component

### PR DESCRIPTION
**Pull request by gfellerph**
Added the radio button and checkbox card component.

### CODE REVIEW

File: packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Radiobutton cards can be grouped together. Use a `fieldset`/`legend` combination to label the group. If there is an error, link the legend with the error message through the `aria-describedby` attribute on the `legend`, pointing to the `id` of the error message.
+
+<Canvas of={RadiobuttonCardStories.Group} />
+
+## SCSS Imports
+
+To import all styles from the design-system-styles package:
+
+```css dark
```
Comment (imagoiq) [inline packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.docs.mdx:28]:
```suggestion
```scss
```

File: packages/documentation-v7/src/stories/components/choice-card/checkbox-card.docs.mdx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+Checkbox cards can be grouped together. Use a `fieldset`/`legend` combination to label the group. If there is an error, link the legend with the error message through the `aria-describedby` attribute on the `legend`, pointing to the `id` of the error message.
+
+<Canvas of={CheckboxCardStories.Group} />
+
+## SCSS Imports
+
+To import all styles from the design-system-styles package:
+
+```css dark
```
Comment (imagoiq) [inline packages/documentation-v7/src/stories/components/choice-card/checkbox-card.docs.mdx:28]:
```suggestion
```scss
```

File: packages/documentation-v7/package.json [hunk 3]
```diff
@@ -42,6 +42,7 @@
     "@storybook/addon-essentials": "7.1.1",
     "@storybook/addon-links": "7.1.1",
     "@storybook/blocks": "7.1.1",
+    "@storybook/client-api": "6.5.16",
```
Comment (imagoiq) [inline packages/documentation-v7/package.json:45]: v6.5 sounds suspicious ?! Isn't there a v7 version?

Comment (gfellerph) [reply packages/documentation-v7/package.json:45]: You're right, it's superseded by @storybook/preview-api. That package is not documented as of now and I did not use either of them anyway (wanted to sync checked status with the control but it's a little complicated so I left it as is for the moment. It would be nice to have but not critical IMHO).

File: packages/documentation-v7/src/stories/components/choice-card/choice-control.snapshot.stories.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  description: ['A very long running description that is wrapping to two lines'],
+  showIcon: [false, true],
+})
+  // Filter out disabled and invalid combinations
+  .filter(args => !(args.disabled && args.validation === 'is-invalid'));
+
+export const CheckboxCard: StoryObj = {
+  render: () => {
+    return html`
+      <div className="d-flex gap-3 flex-wrap">
```
Comment (imagoiq) [inline packages/documentation-v7/src/stories/components/choice-card/choice-control.snapshot.stories.ts:27]:
```suggestion
      <div class="d-flex gap-3 flex-wrap">
```

File: packages/styles/src/components/choice-control-card.scss [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  color: color.$gray-80;
+  background-color: color.$white;
+  border: 2px solid color.$gray-60;
+  border-radius: commons.$border-radius;
+  padding: spacing.$size-regular;
+  transition: color 100ms ease-in-out, background-color 100ms ease-in-out,
+    border-color 100ms ease-in-out;
+
+  // Checked
+  &:where(:has(input:checked)) {
```
Comment (imagoiq) [inline packages/styles/src/components/choice-control-card.scss:24]: :has is not supported by Firefox at the moment.

**Review by imagoiq — CHANGES_REQUESTED** [review]

File: packages/documentation-v7/src/stories/components/choice-card/choice-control.ts [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        id="${id}"
+        name="${args.type}-button-card"
+        class="${inputClasses}"
+        type="${args.type}"
+        disabled="${args.disabled || nothing}"
+        checked="${args.checked || nothing}"
+      />
+      <label id="label-${id}" class="form-check-label" for="${id}">
+        <span>${args.label}</span>
+        ${args.showDescription ? description : null}
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/choice-card/choice-control.ts:141]:
```suggestion
        ${args.showDescription ? description : nothing}
```

File: packages/documentation-v7/src/stories/components/choice-card/choice-control.ts [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+      </label>
+      ${args.showIcon ? icon : null}
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/choice-card/choice-control.ts:143]:
```suggestion
      ${args.showIcon ? icon : nothing}
```

File: packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.docs.mdx [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import { Meta, Canvas, Controls } from '@storybook/blocks';
+import * as RadiobuttonCardStories from './radiobutton-card.stories';
+
+<Meta of={RadiobuttonCardStories} />
+
+# Radio button card
+
+<div className="lead">
+  A larger, more consumer oriented, touch friendly variant of the checkbox used for e.g. selecting
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.docs.mdx:9]:
```suggestion
  A larger, more consumer oriented, touch friendly variant of the radio-button used for e.g. selecting
```

File: packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+export default {
+  ...choiceCardMeta,
+  title: 'Components/Radio Button Card',
+  parameters: {
+    badges: [BADGE.NEEDS_REVISION],
+  },
+};
+
+export const Default = {
+  render: choiceCardDefault,
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts:13]:
```suggestion
  ...choiceCardDefault,
```

File: packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  args: { ...choiceCardMeta.args, type: 'radio' },
+};
+
+export const Group = {
+  render: choiceCardGroup,
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts:18]:
```suggestion
  ...choiceCardGroup,
```

File: packages/styles/src/variables/_type.scss [hunk 11]
```diff
@@ -77,6 +77,7 @@ $font-sizes: (
 );
 
 $font-size-map: (
+  12: $font-size-12,
```
Comment (oliverschuerch) [inline packages/styles/src/variables/_type.scss:80]: Should be done in another PR?

File: packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+export default {
+  ...choiceCardMeta,
+  title: 'Components/Checkbox Card',
+  parameters: {
+    badges: [BADGE.NEEDS_REVISION],
+  },
+};
+
+export const Default = {
+  render: choiceCardDefault,
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts:13]:
```suggestion
  ...choiceCardDefault,
```

File: packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+  args: { ...choiceCardMeta.args, type: 'checkbox' },
+};
+
+export const Group = {
+  render: choiceCardGroup,
```
Comment (oliverschuerch) [inline packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts:18]:
```suggestion
  ...choiceCardGroup,
```

**Review by oliverschuerch — CHANGES_REQUESTED** [review]

**Review by oliverschuerch — COMMENTED** [review]
If a user checks/unchecks the checkbox in the preview story, the control for toggling the `Checked` state is not working anymore.

**Review by oliverschuerch — CHANGES_REQUESTED** [review]
Validation states are only reflected with color. There should be at least a hidden text which tells the user, the field is invalid.

**Review by oliverschuerch — CHANGES_REQUESTED** [review]
HCM Styles are not well.
Use smaller borders for cards and higlight color for selected cards.

File: packages/styles/src/components/choice-control-card.scss (hunk 5 above)
Comment (gfellerph) [reply packages/styles/src/components/choice-control-card.scss]: Added a fallback with classes until it is.

[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/styles/src/components/choice-control-card.scss]: `:where` also reduces specificity to zero which is handy if you want to override with a weaker selector down the line.

File: packages/styles/src/components/choice-control-card.scss [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+    outline-offset: 2px;
+    border-color: color.$black;
+
+    input {
+      border-color: color.$black;
+    }
+  }
+
+  // Hover
+  &:where(:hover:not(:has(input[disabled]))) {
```
[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/styles/src/components/choice-control-card.scss]: ::::::::::::: Updated! :::::::::::::::::

File: packages/styles/src/components/choice-control-card.scss [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+  // Error
+  &:has(input[aria-invalid]),
+  &:has(input.is-invalid) {
+    color: color.$error;
+    border-color: color.$error;
+
+    ~ .invalid-feedback {
+      display: block;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/styles/src/components/choice-control-card.scss]: Adressed

File: packages/styles/src/components/choice-control-card.scss [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+  border-radius: commons.$border-radius;
+  padding: spacing.$size-regular;
+  transition: color 100ms ease-in-out, background-color 100ms ease-in-out,
+    border-color 100ms ease-in-out;
+
+  // Checked
+  &:where(:has(input:checked)) {
+    --post-card-select--hover-bg: #{color.$yellow};
+    background-color: color.$yellow;
+    border-color: color.$gray-60;
```
[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/styles/src/components/choice-control-card.scss:28]: Adressed

File: packages/documentation-v7/src/stories/components/choice-card/choice-control.ts [hunk 17]
```diff
@@ -0,0 +1,168 @@
...
+      table: {
+        category: 'Icon',
+      },
+    },
+  },
+};
+
+export const choiceCardDefault = (args: Args) => {
+  const inputClasses = ['form-check-input', args.validation].filter(c => c !== 'null').join(' ');
+  const id = `control-${crypto.randomUUID().slice(0, 8)}`;
```
[the remark this answers is not in the document]
Comment (gfellerph) [reply packages/documentation-v7/src/stories/components/choice-card/choice-control.ts:146]: used an int counter as it's easier to read for devs using the code snippet

File: packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts (hunk 9 above)
Comment (gfellerph) [reply packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts:13]: choiceCardDefault is a simple function, not a story. This would not work.

File: packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts (hunk 10 above)
Comment (gfellerph) [reply packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts:18]: choiceCardDefault is a simple function, not a story. This would not work.

File: packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts (hunk 12 above)
Comment (gfellerph) [reply packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts:13]: no

File: packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts (hunk 13 above)
Comment (gfellerph) [reply packages/documentation-v7/src/stories/components/choice-card/checkbox-card.stories.ts:18]: also no

**Comment by gfellerph** [comment]
> HCM Styles are not well. Use smaller borders for cards and higlight color for selected cards.

Updated the hover styles, but I would not use the selectedItem color as background, I think it's a bit disruptive to the otherwise very simple HCM.

File: packages/styles/src/components/choice-control-card.scss (hunk 16 above)
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/styles/src/components/choice-control-card.scss:28]: It's not working:
image
image

File: packages/styles/src/components/choice-control-card.scss (hunk 15 above)
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/styles/src/components/choice-control-card.scss]: Not working when checked.
Figma:
image
Storybook:
image

File: packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts (hunk 9 above)
Comment (oliverschuerch) [reply packages/documentation-v7/src/stories/components/choice-card/radiobutton-card.stories.ts:13]: Sorry, my fault!

File: packages/styles/src/components/choice-control-card.scss [hunk 18]
```diff
@@ -0,0 +1,133 @@
+@use '../variables/color';
+@use '../variables/commons';
+@use '../variables/spacing';
+
+.radio-button-card,
+.checkbox-button-card {
+  --post-card-select--hover-bg: #{color.$gray-10};
```
[the remark this answers is not in the document]
Comment (oliverschuerch) [reply packages/styles/src/components/choice-control-card.scss:8]: I'm closing this conversation, as everything is working as expected.
